site/_js/firebase.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * @fileoverview Firebase integration.
+ */
+import {initializeApp} from 'firebase/app';
+//import { initializeAppCheck, ReCaptchaV3Provider } from "firebase/app-check";
+import {getDatabase, ref, runTransaction, onValue} from 'firebase/database';
+
+export class Firebase {
+  constructor() {
+    this.config = null;
+    // For local Firebase Emulator
+    if (location.hostname === 'localhost') {
+      this.config = {
+        databaseURL: 'http://localhost:9000/?ns=oct-sip-test2',
+      };
+      // this.config = {
+      //   apiKey: "AIzaSyB1zPD5QjfGypRydnHE-F6vWaROlHZebSU",
+      //   authDomain: "oct-sip-test2.firebaseapp.com",
+      //   databaseURL: "https://oct-sip-test2-default-rtdb.firebaseio.com",
+      //   projectId: "oct-sip-test2",
+      //   storageBucket: "oct-sip-test2.appspot.com",
+      //   messagingSenderId: "802581884830",
+      //   appId: "1:802581884830:web:54daed54011bd73be6f08f"
+      // }
+    }
+
+    //Initialize firebase app
+    this.firebase = initializeApp(this.config);
+    // this.appCheck = initializeAppCheck(this.firebase, {
+    //   provider: new ReCaptchaV3Provider('6LdHHNweAAAAAA6jtgdZoZ6AFHC2_R4bSnAKY6Fp'),
+    //   isTokenAutoRefreshEnabled: true
+    // });
+    //this.appCheck.activate();
+
+    //Connect to Database
+    this.db = getDatabase(this.firebase);
+    //Getting chrome100 database
+    this.cardsRef = ref(this.db, 'chrome100');
+    //this.cardsRef = ref(this.db,'oct-sip-test2-default-rtdb');
+
+    this.updateLikesCount = this.updateLikesCount.bind(this);
+    this.initialize();
+    this.readData(this.cardsRef);
+  }
+
+  initialize() {
+    const buttons = document.querySelectorAll('.heart-logo');
+    const self = this;
+    for (const button of buttons) {
+      button?.addEventListener('click', function () {
```

Comment:
Rather than adding 100 event listeners, it would be preferable to use event delegation at the container level to check if the click's target had the `heart-logo` class, and if so, execute the callback.

Reply:
Done